### PR TITLE
VV Fixes

### PR DESCRIPTION
**Pull request by Bod9001**
fixes hierarchy refreshing,
Hierarchy now handles objects being deleted

### CODE REVIEW

**Comment by Bod9001** [comment]
let me do quick test

File: UnityProject/Assets/Scripts/Variable Viewer/VariableViewerNetworking.cs [hunk 1]
```diff
@@ -389,7 +389,11 @@ public static List<NetFriendlyHierarchyBookShelf> ProcessLibrary( Librarian.Libr
 	{
 		foreach (var Children in LibraryBookShelf.Contains)
 		{
-			List.Add(ProcessLibraryBookShelf(Librarian.TransformToBookShelf[Children]));
+			if (Librarian.TransformToBookShelf.ContainsKey(Children) == false)
+			{
+				Logger.Log("oph nmananan");
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Variable Viewer/VariableViewerNetworking.cs]: Needed? :p

Comment (pyrosshade) [reply UnityProject/Assets/Scripts/Variable Viewer/VariableViewerNetworking.cs]: what does it meeeaaaaan

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Variable Viewer/VariableViewerNetworking.cs]: oh yeah

**Pull request merged**